.github/workflows/ci.yml

```diff
@@ -38,7 +38,8 @@ jobs:
         run: cargo build
 
       - name: Tests
-        run: cargo test
+        # there should be a unique test thread for native provider tests (asserting spawned processes count)
+        run: cargo test -- --test-threads 1 
```

Comment:
Maybe we can scope this to not run all our test in a single threaded env.

Reply:
What is the reason of threading all tests like this?

Reply:
The reason is because we need to assert the "dummy_process" is running at some point and in some state (paused/restarted/resumed), and if there are multiple tests asserting this (restart, spawn_node, etc...), it will make the test fails, that is what the command refers to.

The performance loss is not that much given the 89 tests of configuration crate run in 0.02s. Only the provider tests takes a bit longer because we assert logs are being/not being written at some point for some time (5/10s) or a process is not present/has restarted (5/10s), the provider tests takes around 1 min.

It takes around 1m30 to run all tests on my machine.

I have a todo to optimize provider tests to run same process with different name.

Reply:
Fixed